webview-ui/src/components/settings/LanguageOptions.tsx

```diff
@@ -1,20 +1,23 @@
 import { VSCodeDropdown, VSCodeOption } from "@vscode/webview-ui-toolkit/react"
 import { memo } from "react"
 import { useTranslation } from "react-i18next"
+import { useExtensionState } from "../../context/ExtensionStateContext"
 
 const LanguageOptions = () => {
-	const { t, i18n } = useTranslation("translation", { keyPrefix: "settingsView", useSuspense: false })
+	const { setCustomInstructions } = useExtensionState()
+	const { t, i18n } = useTranslation("translation")
 
 	const changeLanguage = (e: any) => {
 		const language = e.target.value
 		i18n.changeLanguage(language)
+		setCustomInstructions(t("appRoot.customInstructions"))
```

Comment:
the user may have existing custom instructions that we need to respect. Let's append to existing custom instructions instead.